cmd/agent/process_agent.go

```diff
@@ -0,0 +1,26 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build !windows && bundle_process_agent
+
+// Main package for the agent binary
+package main
+
+import (
+	"os"
+
+	processcommand "github.com/DataDog/datadog-agent/cmd/process-agent/command"
+	processsubcommands "github.com/DataDog/datadog-agent/cmd/process-agent/subcommands"
+	"github.com/DataDog/datadog-agent/pkg/util/flavor"
+	"github.com/spf13/cobra"
+)
+
+func init() {
```

Comment:
I particularly like the idea of not relying on using `init()` and the fact that we can have a single place where all the agent processes are registered, as stated in: 

> This has the advantage that the list of supported agents is explicit in the main.

+ The portability 🎉

Reply:
> The user should not forget to register the Agent even when using init

In this case, forgetting to register would mean completely breaking an agent (such as the process-agent or the security-agent) so this will likely be detected by the CI and our e2e tests.

> This has the advantage that the list of supported agents is explicit in the main.
The thing is that the list is not fixed and depends on build tags, that is why the registration is not done in a single place. This is done so that we have an very easy way to exclude an agent from the single agent binary.

Reply:
> The thing is that the list is not fixed and depends on build tags, that is why the registration is not done in a single place. This is done so that we have an very easy way to exclude an agent from the single agent binary.

In this case my suggestion can be adapted:
```go
func GetProcessAgentCommand() cobra.Command {
   // ...
}
```
become
```
func RegisterCommand(commands map[string]func() *cobra.Command{}) cobra.Command {
   // ...
}
```
`RegisterCommand` will have a different implements with and without build tag. 
Note: we have recently several issues with both global and init.

Reply:
So we need to have 2 files for each service:

- RegisterSecurityAgentCommand in security_agent.go
- An other one that does nothing in no_security_agent.go

If that's ok, I can make the change

Reply:
As discussed offline, we decided that is one of the very few legit ways of registration in `init` functions mostly because it's in the `main` package - and therefore can not be reused by other packages - so there is no risk of side effects as there could be with collector registration for instance.
Having explicit registration requires to add 4 new files, each with one register{Process,SecurityAgent,Trace,SystemProbe} with one implementation that does nothing, which would be confusing.